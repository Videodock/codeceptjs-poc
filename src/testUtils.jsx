import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
