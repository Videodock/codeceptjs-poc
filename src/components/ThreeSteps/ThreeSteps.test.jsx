import React from 'react';
import { render } from '../../testUtils';
import ThreeSteps from './ThreeSteps';

describe('<ThreeSteps />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ThreeSteps />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
