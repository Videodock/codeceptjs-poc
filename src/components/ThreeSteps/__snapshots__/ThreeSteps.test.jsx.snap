// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThreeSteps /> renders and matches snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <section>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-stepsTextContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-stepsTextContent-4"
          >
            <span
              class="makeStyles-number-5"
            >
              1
            </span>
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Pick a score and invite your collaborators
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-stepsImageContainer-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <img
            alt="Pick a score and invite your collaborators"
            src="pick_a_score.png"
          />
        </div>
      </div>
    </section>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <section>
      <div
        class="MuiGrid-root makeStyles-stepWithGradient-6 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-stepsImageContainer-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <img
            alt="Our algorithms will distribute tasks to everyone"
            src="algorithms_will_distribute.png"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-stepsTextContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-stepsTextContent-4"
          >
            <span
              class="makeStyles-number-5"
            >
              2
            </span>
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Our algorithms will distribute tasks to everyone
            </h2>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <section>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-stepsTextContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-stepsTextContent-4"
          >
            <span
              class="makeStyles-number-5"
            >
              3
            </span>
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Every collaborator can contribute to a better score
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-stepsImageContainer-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <img
            alt="Every collaborator can contribute to a better score"
            src="every_collaborator.png"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
